instrumentation/jdbi3/src/main/java/brave/jdbi3/TracingSqlLogger.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenZipkin Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package brave.jdbi3;
+
+import brave.Span;
+import brave.Tracing;
+import brave.propagation.ThreadLocalSpan;
+import java.net.URI;
+import java.sql.SQLException;
+import org.jdbi.v3.core.statement.SqlLogger;
+import org.jdbi.v3.core.statement.SqlStatements;
+import org.jdbi.v3.core.statement.StatementContext;
+
+import static brave.Span.Kind.CLIENT;
+
+final class TracingSqlLogger implements SqlLogger {
+  final Tracing tracing;
+  final ThreadLocalSpan threadLocalSpan;
+  final String remoteServiceName;
+
+  TracingSqlLogger(JdbiTracing jdbiTracing) {
+    this.tracing = jdbiTracing.tracing;
+    this.remoteServiceName = jdbiTracing.remoteServiceName;
+    this.threadLocalSpan = ThreadLocalSpan.create(tracing.tracer());
+  }
+
+  @Override public void logBeforeExecution(StatementContext ctx) {
+    Span span = threadLocalSpan.next().kind(CLIENT);
+    if (span == null || span.isNoop()) return;
+    ctx.setTraceId(span.context().traceIdString());
+
+    final String renderedSql = ctx.getRenderedSql();
+    if (renderedSql != null) {
+      String sql = renderedSql.substring(
+        0,
+        Math.min(renderedSql.length(), ctx.getConfig(SqlStatements.class).getJfrSqlMaxLength())
+      );
+      int spaceIndex = sql.indexOf(' '); // Allow span names of single-word statements like COMMIT
+      span.name(spaceIndex == -1 ? sql : sql.substring(0, spaceIndex));
+      span.tag("sql.query", sql);
+    } else {
+      span.name(ctx.describeJdbiStatementType());
+    }
+
+    span.remoteServiceName(remoteServiceName);
+    if (ctx.getConnection() != null) {
+      try {
+        String jdbcUrl = ctx.getConnection().getMetaData().getURL();
+        parseServerIpAndPort(jdbcUrl, span, remoteServiceName == null);
+      } catch (SQLException ignored) {
+      }
+    }
+
+    span.start();
+  }
+
+  @Override public void logAfterExecution(StatementContext ctx) {
+    Span span = threadLocalSpan.remove();
+    if (span == null) return;
+    span.finish();
+  }
+
+  @Override public void logException(StatementContext ctx, SQLException ex) {
+    Span span = threadLocalSpan.remove();
+    if (span == null) return;
+    span.error(ex);
+    span.finish();
+  }
+
+  static void parseServerIpAndPort(String jdbcUrl, Span span, boolean addRemoteServiceName) {
+    // only allocate a URI once for both service name and remote address
+    URI url = URI.create(jdbcUrl.substring(5)); // strip "jdbc:"
```

Comment:
Can consider "jdbc:".length instead of 5 since then I think don't need the comment.

Also just a double check that we don't need to actually check the prefix before trimming